packages/opentelemetry-resources/src/platform/browser/detect-resources.ts

```diff
@@ -47,3 +49,51 @@ export const detectResources = async (
     Resource.empty()
   );
 };
+
+/**
+ * Runs all resource detectors synchronously, merging their results. Any asynchronous
+ * attributes will be merged together in-order after they resolve.
+ *
+ * @param config Configuration for resource detection
+ */
+export const detectResourcesSync = (
+  config: ResourceDetectionConfig = {}
+): Resource => {
+  const internalConfig: ResourceDetectionConfig = Object.assign(config);
+
+  const resources: Resource[] = (internalConfig.detectors ?? []).map(d => {
+    try {
+      const resourceOrPromise = d.detect(internalConfig);
+      let resource: Resource;
+      if (resourceOrPromise instanceof Promise) {
```

Comment:
Typically, we should support all `thenable`s instead of check if the prototype of the value is the global Promise.

```suggestion
      if (typeof resourceOrPromise.then === 'function') {
```

Reply:
I don't think this is allowed as `resourceOrPromise` can also be of type `Resource`.

Reply:
`Resource` doesn't have a `then` method. It is not a thenable anyways. To circumvent the typescript complaint, we can do this with type narrowing:

```ts
function isPromiseLike<R>(val: any): val is PromiseLike<R> {
  return val !== null && typeof val === 'object' && typeof val.then === 'function';
}

if (isPromiseLike<Resource>(resourceOrPromise)) {
  resourceOrPromise.then() // <= we can now safely use the `then` method.
}
```